lib/datadog/profiling/component.rb

```diff
@@ -162,32 +148,70 @@ def self.build_profiler_component(settings:, agent_settings:, optional_tracer:)
           return false
         end
 
-        return true if settings.profiling.advanced.force_enable_new_profiler
+        true
+      end
+
+      private_class_method def self.no_signals_workaround_enabled?(settings) # rubocop:disable Metrics/MethodLength
+        setting_value = settings.profiling.advanced.no_signals_workaround_enabled
+        legacy_ruby_that_should_use_workaround = RUBY_VERSION.start_with?('2.3.', '2.4.', '2.5.')
+
+        unless [true, false, :auto].include?(setting_value)
+          Datadog.logger.error(
+            "Ignoring invalid value for profiling no_signals_workaround_enabled setting: #{setting_value.inspect}"
```

Comment:
Should we tell which are the valid values?

Reply:
Changed in 69fce3cd39493c30545f23698ca1e42f8193bb36